pkgs/development/compilers/openjdk/darwin/18.nix

```diff
@@ -0,0 +1,81 @@
+{ lib, stdenv, fetchurl, unzip, setJavaClassPath }:
+let
+  # Details from https://www.azul.com/downloads/?version=java-18-sts&os=macos&package=jdk
+  # Note that the latest build may differ by platform
+  dist = {
+    x86_64-darwin = {
+      arch = "x64";
+      zuluVersion = "18.28.13";
+      jdkVersion = "18.0.0";
+      sha256 = "0hc5m3d4q3n7sighq3pxkdg93vsrgj1kzla1py9nfnm9pnj9l2kq";
+    };
+
+    aarch64-darwin = {
+      arch = "aarch64";
+      zuluVersion = "18.28.13";
+      jdkVersion = "18.0.0";
+      sha256 = "0ch4jp2d4pjvxbmbswvjwf7w2flajrvjg5f16ggiy80y8l0y15cm";
+    };
+  }."${stdenv.hostPlatform.system}";
```

Comment:
Missing or throw unsupported system or eval fails on other platforms.

Reply:
With all respect, it a. doesn't get called for non-Darwin platforms, and b. is the same way that all OpenJDK derivations have done this. I can change it if required, but it seems unnecessary and out of line.

https://github.com/NixOS/nixpkgs/blob/992d767bfd0fefa33426d36342830b5d2588a132/pkgs/top-level/java-packages.nix#L51-L56
https://github.com/NixOS/nixpkgs/blob/a6b757a52bf2370540ffc2c0210e25e79b419a63/pkgs/top-level/java-packages.nix#L187-L193

Reply:
Thats not great. It should at least eval on any platform otherwise you can't check if your changes on break evaluation for another platform.

Reply:
Wouldn't you set the `system` anyway when testing evaluation for another platform, then `stdenv.hostPlatform` would show that?

Anyway I don't think we put too many expectations on the evaluation of packages if they're not checked on hydra / ofborg, and all the ofborg builds on the PR are green, so I think it's ok.

Reply:
> I don't think we put too many expectations on the evaluation of packages if they're not checked on hydra / ofborg

Especially important to:
1. Help reviewers be more effective, since they can rely on ofborg builds and focus on other aspects of the PR.
2. Help contributors get faster signal on their PRs, and have a higher confidence that their code is good if the build is green.